presto-main/src/main/java/io/prestosql/sql/SqlFormatterUtil.java

```diff
@@ -37,9 +40,11 @@ public static String getFormattedSql(Statement statement, SqlParser sqlParser)
             parsed = sqlParser.createStatement(sql, parsingOptions);
         }
         catch (ParsingException e) {
-            throw new PrestoException(GENERIC_INTERNAL_ERROR, "Formatted query does not parse: " + statement);
+            log.debug("Formatted query does not parse: \n\t\t statement: [%s] \n\t\t formatted: [%s]", statement, sql);
+            throw new PrestoException(GENERIC_INTERNAL_ERROR, "Formatted query does not parse: " + statement, e);
```

Comment:
Since statement is likely huge, it might be better to attach, which also gives us a place to put the SQL:
```java
PrestoException error = new PrestoException(GENERIC_INTERNAL_ERROR, "Formatted query does not parse", e);
error.addSuppressed(new RuntimeException("Statement: " + statement));
error.addSuppressed(new RuntimeException(format("Formatted: [%s]", sql)));
throw error;
```

Reply:
nice, i can do that

note that sql will contain many newlines anyway (same for statement to string), which will affect any fmting we do

(i don't want to reprocess the sql to add indents though)

Reply:
I now realize that if we include all the information in the exception, we shouldn't log, since that violates the no log-and-throw principal. (i.e. all the info is available in the query failure, just like for any other failure)

Reply:
that's why i removed logging